### ISSUE TITLE
Missing MsalClientException.NetworkNotAvailableError on iOS & Android

### ISSUE THREAD

**Issue by KarinBerg** [AzureAD/microsoft-authentication-library-for-dotnet#592]
For platform Xamarin.iOS and Xamarin.Android there is no check for an available network connection in class WebUI. So in this case an HttpRequestException is thrown and not the expected MsalClientException with ErrorCode "NetworkNotAvailableError".

Please add an network connection check for this platforms to be consistent over all platforms.

**Comment by bgavrilMS** [comment]
+ @jmprieur for triage

**Comment by henrik-me** [comment]
Will require a bit of refactoring. I assume this is possible especially with the latest updates in the dev branch. @bgavrilMS can you pls. take a look at this?

**Comment by henrik-me** [comment]
@bgavrilMS : still relevant? Putting this in 4.3.
@jmprieur : fix or close? (#Supportability)

**Comment by bgavrilMS** [comment]
I had a look at this and I propose we fix it by removing the custom exception from .net (this is a breaking behavioural change though), but then again so is the proposed change.

The reasons: 

- Network is down exceptions are fairly common and app developers should treat them. Even a generic "Something went wrong" message would do. 

- MSAL makes many calls to the network, such as discovery calls. It would be a perf hit to try to read the state of the network before each HTTP call. 

- Catching HTTP exceptions and repackaging them would also be a breaking change to our 

- Detecting that the network is down is tricky - see how it's been done in Xamarin Essentials - https://github.com/xamarin/Essentials/blob/master/Xamarin.Essentials/Connectivity/Connectivity.android.cs - also note that it does not guarantee that a device has Internet, it just shows that a connection is available. 

All in all, I don't think it's MSAL's responsibility to provide internet detection capabilities. App developers should not have to have special catch blocks for MSAL. 

This is easier: 

```csharp
try
{
   // call MSAL
   // call some other logic that makes HTTP calls
}
catch HttpException
  // Handle all unexpected exceptions, including HTTP issues
```

Than custom handling: 


```csharp
try
{
   // call MSAL
   // call some other logic that makes HTTP calls
}
catch (MsalClientException ex) when ex.ErrorCode == NetworkIsDown
{
   // handle MSAL
}
catch HttpException
{
 // handle all other Http issues
}
```

**Comment by jmprieur** [comment]
This makes sense to me.
Let's provide guidance, then. Let's show this in our sample and the Mobile scenario landing page, as well as the error handling wiki/docs.ms page?

**Comment by henrik-me** [comment]
I'm ok with this. Only reason to consider leaving is the consideration for a major version bump.

**Comment by henrik-me** [comment]
Fix for consistency means the library will not handle network down type of errors.

**Comment by jennyf19** [comment]
Included in [4.3.0](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.3.0) release
